crates/bevy_animation/src/lib.rs

```diff
@@ -111,12 +88,71 @@ pub struct VariableCurve {
     /// - for `Interpolation::Step` and `Interpolation::Linear`, each keyframe is a single value
     /// - for `Interpolation::CubicSpline`, each keyframe is made of three values for `tangent_in`,
     ///     `keyframe_value` and `tangent_out`
-    pub keyframes: Keyframes,
+    pub keyframes: Box<dyn Keyframes>,
     /// Interpolation method to use between keyframes.
     pub interpolation: Interpolation,
 }
 
+impl Clone for VariableCurve {
+    fn clone(&self) -> Self {
+        VariableCurve {
+            keyframe_timestamps: self.keyframe_timestamps.clone(),
+            keyframes: Keyframes::clone_value(&*self.keyframes),
+            interpolation: self.interpolation,
+        }
+    }
+}
+
 impl VariableCurve {
+    /// Creates a new curve from timestamps, keyframes, and interpolation type.
+    ///
+    /// The two arrays must have the same length.
+    pub fn new<K>(
+        keyframe_timestamps: Vec<f32>,
+        keyframes: impl Into<K>,
+        interpolation: Interpolation,
+    ) -> VariableCurve
+    where
+        K: Keyframes,
+    {
+        VariableCurve {
+            keyframe_timestamps,
+            keyframes: Box::new(keyframes.into()),
+            interpolation,
+        }
+    }
+
+    /// Creates a new curve from timestamps and keyframes with no interpolation.
+    ///
+    /// The two arrays must have the same length.
+    pub fn step<K>(keyframe_timestamps: Vec<f32>, keyframes: impl Into<K>) -> VariableCurve
+    where
+        K: Keyframes,
+    {
+        VariableCurve::new(keyframe_timestamps, keyframes, Interpolation::Step)
+    }
```

Comment:
Maybe we could change `keyframe_timestamps` to accept `impl Into<Vec<f32>>` to allow calling this using an array?

```suggestion
    pub fn step<K>(
        keyframe_timestamps: impl Into<Vec<f32>>,
        keyframes: impl Into<K>,
    ) -> VariableCurve
    where
        K: Keyframes,
    {
        VariableCurve::new(keyframe_timestamps.into(), keyframes, Interpolation::Step)
    }
```

Reply:
Ok, I made that change.